inst/private/doctest_collect.m

```diff
@@ -125,6 +125,19 @@
 end
 
 
+% Deal with .oct files, which may define more than one function
+if (strcmp(type, 'octfile'))
+  len = numel (what) + 4;
+  list = autoload ();
+  pmatch = @(e) (numel (e.file) > len) && strcmp (e.file(end-len+1:end), [what '.oct']);
+  idx = find (arrayfun (pmatch, list));
```

Comment:
this arrayfun approach probably nicer than my approach!  Nice shortcut with numel.